### ISSUE TITLE
Record at Native Resolution

### ISSUE THREAD

**Issue by Xrot** [Genymobile/scrcpy#4047]
My android phone is 1080x2400 natively, but for some reason scrcpy when recording it records on 864x1920 resolution.
How can i make it record a video on native resolution

**Comment by rom1v** [comment]
Probably because your device encoder could not encode at the native screen resolution. Look at the output in the console on start (and see `--no-downsize-on-error`).

**Comment by Xrot** [comment]
it's probably an encoder, my phone is redmi note 11, what encoder should i make it use instead? idk what it is using currently

**Comment by rom1v** [comment]
You can list your "encoders" with `--list-encoders` (see https://github.com/Genymobile/scrcpy/blob/master/doc/video.md#codec).

But if your hardware encoder could not encode above 1920, you will not be able to capture your screen at 1080x2400.

**Comment by Xrot** [comment]
thank you will try

**Comment by Xrot** [comment]
![image](https://github.com/Genymobile/scrcpy/assets/25128553/7abb0ded-5549-4038-813d-b2679ea1b0fb)

**Comment by Xrot** [comment]
Which one of these will work ok with Vegas Pro 20 if you know?

**Comment by Xrot** [comment]
Thank you very much! it is now recording on native resolution, the OMX.qcom.video.encoder.avc was bugging out

**Comment by Xrot** [comment]
is this something of concern? WARN: [FFmpeg] track 1: codec frame size is not set

**Comment by Xrot** [comment]
Also, is there a way to make recording performance as fast as possible? what if i do --no-display will that even do anything

**Comment by rom1v** [comment]
> is this something of concern? WARN: [FFmpeg] track 1: codec frame size is not set

Nope, it is a known warning related to muxing opus without knowing the frame size in advance (but Android does not give it explicitly).

> Also, is there a way to make recording performance as fast as possible?

Do you experience any performance issue? It should record every frame produced by the device. If it is not the case, then your hardware encoder is not effcient enough.

**Comment by Xrot** [comment]
Just wanted to limit FPS to 30 and somehow make frames very smooth. Without smoothing 30fps looks stuttery sometimes

**Comment by Xrot** [comment]
but my phone can record at 60fps without problem

**Comment by rom1v** [comment]
> Just wanted to limit FPS to 30 and somehow make frames very smooth

https://github.com/Genymobile/scrcpy/blob/master/doc/video.md#frame-rate

> Without smoothing 30fps looks stuttery sometimes

In the scrcpy window, or when you play the recorded file (or both)?
Over USB or Wifi?

**Comment by Xrot** [comment]
When i play recorded file, USB

**Comment by Xrot** [comment]
Do i really have to record at 60fps for 30fps in video editing software to look smooth? Maybe it's possible while recording with scrcpy no?

**Comment by Xrot** [comment]
My issue is that, Vegas Pro 20 doesn't like Opus and OMX encoders, Is there a way to install more encoders on my android device, or does scrcpy have a ffmpeg integrated so i can re-encode my recordings?

**Comment by rom1v** [comment]
> My issue is that, Vegas Pro 20 doesn't like Opus and OMX encoders, Is there a way to install more encoders on my android device

The video encoders use an hardware encoder. You can't change it (if it is not efficient enough), it depends on the device (typically high-end devices have better encoders).

> does scrcpy have a ffmpeg integrated so i can re-encode my recordings?

You can reencode with ffmpeg on your computer what you recorded with scrcpy.
